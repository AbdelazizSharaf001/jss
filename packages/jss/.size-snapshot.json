{
  "dist/jss.js": {
    "bundled": 77413,
    "minified": 21696,
    "gzipped": 6419
  },
  "dist/jss.min.js": {
    "bundled": 77413,
    "minified": 21696,
    "gzipped": 6419
  }
}
