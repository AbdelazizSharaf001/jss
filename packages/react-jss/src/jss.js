// @flow
import {create} from 'jss'
import preset from 'jss-preset-default'

export const jss = create(preset())
