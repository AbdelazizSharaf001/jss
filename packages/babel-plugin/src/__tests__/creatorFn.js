import {stripIndent} from 'common-tags'
import {sheets} from 'jss'
import {transform} from '../testUtils'

describe('styles creator function', () => {
  beforeEach(() => {
    sheets.reset()
  })

  test('arrow function', () => {
    const code = stripIndent`
      import injectSheet from 'react-jss';
      injectSheet(() => ({
        a: {
          color: 'red'
        }
      }));
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('arrow function with return', () => {
    const code = stripIndent`
      import injectSheet from 'react-jss';
      injectSheet(() => {
        return {
          a: {
            color: 'red'
          }
        };
      });
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('function expression', () => {
    const code = stripIndent`
      import injectSheet from 'react-jss';
      injectSheet(function () {
        return {
          a: {
            color: 'red'
          }
        };
      });
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('function ref', () => {
    const code = stripIndent`
      import injectSheet from 'react-jss';

      function getStyles() {
        return {
          a: {
            color: 'red'
          }
        };
      }

      injectSheet(getStyles);
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('prop access inside styles creator fn', () => {
    const code = stripIndent`
      import injectSheet from 'react-jss';
      const config = {
        primary: 'red'
      };
      injectSheet(() => ({
        a: {
          color: config.primary
        }
      }));
    `
    expect(transform(code)).toMatchSnapshot()
  })

  test('theme prop access over babel config as fn arg', () => {
    const code = stripIndent`
      import injectSheet from 'react-jss';
      injectSheet(theme => ({
        a: {
          color: theme.primary
        }
      }));
    `
    const options = {theme: {primary: 'red'}}
    expect(transform(code, options)).toMatchSnapshot()
  })

  test('complex theme prop access over babel config as fn arg', () => {
    const code = stripIndent`
      import injectSheet from 'react-jss';
      injectSheet(theme => ({
        a: {
          color: theme.x[0].color
        }
      }));
    `
    const options = {theme: {x: [{color: 'red'}]}}
    expect(transform(code, options)).toMatchSnapshot()
  })

  test('throw prop access error complex theme prop access over babel config as fn arg', () => {
    const code = stripIndent`
      import injectSheet from 'react-jss';
      injectSheet(theme => ({
        a: {
          color: theme.x[0].color
        }
      }));
    `
    expect(() => transform(code, {})).toThrowError()
  })
})
