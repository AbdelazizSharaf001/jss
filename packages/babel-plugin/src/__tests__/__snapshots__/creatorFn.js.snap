// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styles creator function arrow function 1`] = `
"import injectSheet from 'react-jss';
injectSheet(() => ({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
}), {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;

exports[`styles creator function arrow function with return 1`] = `
"import injectSheet from 'react-jss';
injectSheet(() => {
  return {
    \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
  };
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;

exports[`styles creator function complex theme prop access over babel config as fn arg 1`] = `
"import injectSheet from 'react-jss';
injectSheet(theme => ({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
}), {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;

exports[`styles creator function function expression 1`] = `
"import injectSheet from 'react-jss';
injectSheet(function () {
  return {
    \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
  };
}, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;

exports[`styles creator function function ref 1`] = `
"import injectSheet from 'react-jss';

function getStyles() {
  return {
    \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
  };
}

injectSheet(getStyles, {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;

exports[`styles creator function prop access inside styles creator fn 1`] = `
"import injectSheet from 'react-jss';
const config = {
  primary: 'red'
};
injectSheet(() => ({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
}), {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;

exports[`styles creator function theme prop access over babel config as fn arg 1`] = `
"import injectSheet from 'react-jss';
injectSheet(theme => ({
  \\"@raw\\": \\".a-id {\\\\n  color: red;\\\\n}\\"
}), {
  \\"classes\\": {
    \\"a\\": \\"a-id\\"
  }
});"
`;
